lua/controller.lua

```diff
@@ -502,6 +519,39 @@ local function register_controller()
 	core.register_node("drawers:controller", def)
 end
 
+function controller_tl_on_pull(pos, side, player_name)
+	if not controller_allow_metadata_inventory_take then return false end
+	local meta = core.get_meta(pos)
+	local drawer_net_index = index_drawers(pos)
+	meta:set_string("drawers_table_index", core.serialize(drawer_net_index))
+	local first = nil
+	for index,_ in pairs(drawer_net_index) do
+		if index and index ~= "empty" then
```

Comment:
Table index cannot be nil.

Reply:
I'm sorry, I don't understand. Do you mean maybe it should be like this?
`local first = {}`

Would this still work with that?
`if first then`

Reply:
Oh, I see now you meant 'if index and index ~= "empty"`
Ok, updated to `"if index ~= "empty"`